First pr

alexnazarv opened the pull request:
Го посмотрим что есть, а остальное надеюсь в субботу успею доделать

Commit:
first commit

Commit:
first day

Commit:
first day

Commit:
first day

Commit:
first day

Commit:
second day

Commit:
second day

Commit:
second day

Commit:
second day

Commit:
day three

Commit:
day four

Commit:
day fourth

Commit:
Day fourth

Commit:
day fourth

Commit:
day fourth

Commit:
day fifth

moneretin commented on setup.cfg:
```diff
@@ -0,0 +1,43 @@
+[isort]
```
Сделал бы один файл с .flake и туда все что с флейк, остальную конфигурацию в pyproject, настройки pytets в pyproject

moneretin commented on setup.cfg:
```diff
@@ -0,0 +1,43 @@
+per-file-ignores =
+# There is only one interface
+    app/main.py: S104,
+    tests/*: DAR101,S101,WPS306,WPS226
```
Комменты что это за проверки?

moneretin commented on Dockerfile:
```diff
@@ -7,5 +7,5 @@ FROM python:3.10.10
 WORKDIR /code
 COPY --from=requirements-stage /tmp/requirements.txt /code/requirements.txt
```
оставляем поетри в имедже, можно замерить размер

moneretin replied:
еще нет dockerignore файла

moneretin commented on README.md:
````diff
@@ -1,5 +1,50 @@
+
+### Run linters
+```bash
+poetry run flake8 . && mypy . && isort .
````
Замени эту команду на pre-commit тулу, позязя
Используй pre-commit-config.yaml

moneretin commented on README.md:
````diff
@@ -1,5 +1,50 @@
+
+#### Build image and run container
+```bash
+docker build . -t apicontainer -t apicontainer:0.1 && 
````
Используй semantic release в своем CI/CD
Когда билдишь вручную, делай тег лейтест для тестирования или тег тест
Для поднятия версий используй конвенчитанл коммит

Commit:
first run

Commit:
day second

Commit:
adjust readme

Commit:
cache poetry

Commit:
quick fix

Commit:
check if cache works

Commit:
check if cache works

Commit:
if cache works

Commit:
adjust read default

Commit:
check

Commit:
check

Commit:
check

Commit:
check

Commit:
check

Commit:
check

Commit:
update pre-commit

Commit:
adjust readme

Commit:
build(Dockerfile,-pyproject.toml,-ci.yml): add commitzen and improve Dockerfile

make image install special version of poetry and add flag --no-root as now we build a python package as well as docker image

Commit:
fix(test_main): change one of the prices in tests

string to see how does it look in changelog

Commit:
bump: version 0.1.0 → 0.1.1

Commit:
fix(models): updated

Commit:
fix(core): updated
fix(model): optimized

Commit:
fix(core): updated

fix(model): optimized

Commit:
fix(core): updated

fix(model): fixed
